src/main/java/com/github/dockerjava/core/command/ListImagesCmdImpl.java

```diff
@@ -48,4 +49,22 @@ public ListImagesCmd withFilters(String filter) {
         return this;
     }
 
+    @Override
+    public String toString() {
+        return new StringBuilder("images ").append(showAll ? "--all=true" : "")
+                .append(filters != null ? "--filters " + filters : "")
+                .append(imageNameFilter != null ? "--filter " + imageNameFilter : "").toString();
+    }
+
+    @Override
+    public ListImagesCmd withImageNameFilter(String imageNameFilter) {
+        checkNotNull(imageNameFilter, "image name filter not specified");
```

Comment:
Maybe it was OK to check in exec as it coudn't "know" that this was checked already. When introducing `@Nonnull` annotation the checkNotNull is not needed anymore.